Orderable Spaces

keithpenrod opened the pull request:
Adding new properties of oderable space, linear continuum, and some results related to them.

prabau commented:
@keithpenrod Can you list here in words which theorems you want add?  That will make it easier to review things.

keithpenrod commented:
> @keithpenrod Can you list here in words which theorems you want add? That will make it easier to review things.



> @keithpenrod Can you list here in words which theorems you want add? That will make it easier to review things.

273. Every ordered space is T_4
274. If X is T_2, compact, and second countable then it is metrizable
275. If X is T_2, compact, and metrizable then it is second countable (274 and 275 are really one theorem, but I didn't know how to do an iff)
276. Every linear continuum is connected and locally connected
277. Every linear continuum is orderable
278. Every linear continuum has distinct points
279. Every separable linear continuum is second countable and metrizable
280. Every metrizable linear continuum is completely metrizable
281. Every linear continuum has cardinality at least as large as the continuum (reals)
282. Every linear continuum is locally compact

prabau commented:
What do theorems 274 and 275 have to do with ordered spaces?  If not , it does not belong it this pull request.

prabau commented:
pi-base is meant as a repository of interesting examples and counterexamples in topology.  We only add theorems to the extent that they allow automatic deduction of properties from other properties, so that the number of properties that need to be explicitly stated for each space can be somewhat reduced.

In this particular instance, proposed theorems 274 and 275 are already a consequence of other results, so adding them would not allow any further deductions or simplifications.  This can be checked for 275 like this for example:
https://topology.pi-base.org/spaces?q=T_2%2BCompact%2BMetrizable%2B~Second%20Countable
and similarly for 274.

keithpenrod commented:
> pi-base is meant as a repository of interesting examples and counterexamples in topology. We only add theorems to the extent that they allow automatic deduction of properties from other properties, so that the number of properties that need to be explicitly stated for each space can be somewhat reduced.
> 
> In this particular instance, proposed theorems 274 and 275 are already a consequence of other results, so adding them would not allow any further deductions or simplifications. This can be checked for 275 like this for example: https://topology.pi-base.org/spaces?q=T_2%2BCompact%2BMetrizable%2B~Second%20Countable and similarly for 274.

That makes sense. I removed 274 and 275. I split 279 into separate theorems (now 274 and 275) after seeing issue #132, and renumbered 282 as 279 for contiguous numbering.

prabau commented:
Sorry, when you get the chance, could you write the updated list of theorems in English here one more time?  The official maintainers of the repo will appreciate it.

keithpenrod commented:
No problem.
273. Every ordered space is T_4
274. Every separable linear continuum is second countable
275. Every separable linear continuum is metrizable
276. Every linear continuum is connected and locally connected
277. Every linear continuum is orderable
278. Every linear continuum has distinct points
279. Every linear continuum is locally compact
280. Every metrizable linear continuum is completely metrizable
281. Every linear continuum has cardinality at least as large as the continuum (reals)

prabau commented:
@keithpenrod Thanks for adding the Orderable property.  That's a nice property to have, as it implies many nontrivial other ones (more on that below).

Comparing the terminology used in graduate level sources, Willard (6D) uses "ordered spaces" (= "a linearly orders set with its ordered topology"); Engelking (1.7.4) has "linearly ordered space" (with the meaning of linearly orderable), as does Arhangel'skii; Encyclopedia of General Topology (2004) has a whole chapter on this and uses "linearly ordered topological space" abbreviated "LOTS", as do many papers.  Based on that, I am wondering if it wouldn't be better to use "Linearly orderable" to better match common usage and hint that it's for linear orders and not partial orders ?
Also would be good to mention the terminology "linearly ordered topology space (LOTS)" in the description, and add LOTS as an alias.
And also would be good to add mention of Steen & Seebach example 39 ("order topology"), as it's a whole section dedicated to this.

Regarding thm 273 (LOTS ==> T4), it's true; but it does not seem that it would add much.  We can instead replace this with the stronger result: LOTS implies T5.  Source: Engelking, 2.7.5, and also mathse:322683 (https://math.stackexchange.com/questions/322683) [worth mentioning in pi-base if we make this a theorem].  Also mention Steen&Seebach example 39, which has a detailed proof of that same result.

It is also a fact that: LOTS implies Collectionwise normal.  That's mentioned in the mathse above.  But an even better result would be: LOTS implies Hereditarily collectionwise normal, as that last property implies both Coll. normal and Completely normal.  The thing is the property "hereditarily collectionwise normal" is not yet in pi-base.  I was reluctant to add it, because there are so many other issues/cleanup that need to be done in the existing pi-base and I did not want to overwhelm the reviewers.  But would be nice to have at some point.

For background info, the direct proof of (LOTS ==> T5) is not easy.  I find going via still another property longer but not too painful: LOTS ==> Monotonically normal ==> Hereditarily collectionwise normal ==> etc.  See the two wikipedia articles and the references therein:
https://en.wikipedia.org/wiki/Monotonically_normal_space
https://en.wikipedia.org/wiki/Collectionwise_normal_space#Hereditarily_collectionwise_normal_space
(though I think adding Monotonically normal to pi-base would be too much at this point)

Will be good to get Steven and Chris's input after we hash this out at much as we can between the two of us.

Ah, and one more thing:  In general, adding a new theorem sometimes allows to simplify pi-base by eliminating existing traits files (that's what they call the property files for various spaces) if the corresponding property can now be deduced instead explicitly stated in traits file.  But the philosophy recommended by Steven and Chris is to keep the redundant traits file around if there is a pedagogical reason to keep some information.  One example where we can safely remove it is when it only quotes the S&S General Reference Chart as justification.

-- end of message, sorry for the length :-)

keithpenrod commented:
Thank you. I knew that linearly ordered spaces were T5 but I didn't have a reference for it and didn't want to try writing up the proof myself. I've made the requested change for the name of "linearly orderable". I've update theorem 273 which now states linearly orderable implies T5. If it would be useful, I'd be happy to add the properties of monotonically normal and hereditarily collectionwise normal.

StevenClontz commented:
> I knew that linearly ordered spaces were T5 but I didn't have a reference for it and didn't want to try writing up the proof myself. 

The direction we're heading for this is to encourage folks to post to Math.SE or MathOverflow (as appropriate): this way we aren't so much in the business of peer review but reflecting peer reviewed mathematics.

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+
+A space whose topology can be induced by a total order on the underlying set.
+
+Defined on page 84 of {{mr:3728284}}, in 6D of {{mr:MR2048350}}, and 1.7.4 of {{mr:MR1039321}}.
```
I think we should also mention S&S, something like "Also defined in counterexample #39 ("Order Topology") in {{doi:10.1007/978-1-4612-6290-9}}."

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+  - Linearly ordered topological space
+  - LOTS
+refs:
+  - wikipedia: Order_topology
+    name: Order topology
```
Move the wikipedia reference to the end of the list, as it's always preferable to have published books or peer reviewed journal articles first.  Also for consistency with the rest of pi-base, we usually add "on Wikipedia" to the end of the name field (see https://topology.pi-base.org/properties/P000090/references for a random example).  Makes it easier for the reader to know it's from Wikipedia.

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+name: Linearly orderable
+aliases:
+  - Linearly ordered topological space
+  - LOTS
```
What do you think of the idea of making LOTS the primary name and making "Linearly orderable" the alias?  Would make it short and easily read in lists of theorems.

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+slug: linearly_orderable
+name: Linearly orderable
+aliases:
+  - Linearly ordered topological space
```
A little long for an alias?  Was thinking more of having this in the description below as an explanation for LOTS. But up to you.

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+---
+uid: P000133
+slug: linearly_orderable
```
Question for myself: Slugs are everywhere.  But what is a slug, what does it do?

prabau commented on properties/P000133.md:
```diff
@@ -0,0 +1,21 @@
+    name: General Topology (Engelking, 1989)
+---
+
+A space whose topology can be induced by a total order on the underlying set.
```
A reader should be able to get the definition just from reading this without having to look up sources. So need to describe what that topology induced by a total order is.

prabau commented:
Regarding the topological notion of "linear continuum", I can't seem to find any reference to this terminology anywhere for topological spaces, except people on mathse solving problems related to Munkres.  Can you?

Actually, rereading Munkres to make sure.  He writes on p. 153: " A simply ordered set L having more than one element is called a linear continuum if ...".  And then states the theorem: "If L is a linear continuum in the order topology, then L is connected ..." (i.e. he is talking about a topological space whose topology is induced by some linear continuum).
Strictly speaking, he never defines _linear continuum_ to be a topological property.  Instead, it's a property of an order relation, belonging to order theory and not to topology (which matches https://en.wikipedia.org/wiki/Linear_continuum).  Of course the topology that it induces has some nice properties.  And in fact, ignoring the fact that it requires at least two points, the notion is identical to a connected LOTS. It's an if and only if.

So wouldn't it make more sense to not introduce this property and instead have various theorems with connected LOTS?
@StevenClontz @ccaruvana @keithpenrod

StevenClontz commented:
Trying to get to this but it's the last couple weeks of the semester and students are diligently using their deserved office hours... :-)

I also prefer to not have a specific "linear continuum" property and instead use [linearly orderable](https://topology.pi-base.org/properties/P000133) + [connected](https://topology.pi-base.org/properties/P000036). In addition to not introducing another property, we also avoid confusing terminology: in modern usage, a "continuum" is a compact connected Hausdorff (or sometimes metric) space (we also don't have a "continuum" property in lieu of combining those three properties), while a "linear continuum" need not be compact.

prabau commented:
Given that, this pull request is already becoming quite long.  Here is an idea: For this pull request, just introduce the LOTS property and add the traits for various spaces that satisfy it.  Keep theorem 273 and leave all the other proposed theorems to later pull requests.  Once it's deployed, it will be nicer to be able to play with it in situ and go from there.

StevenClontz commented:
I like @prabau's suggestion. Closing for now - if @keithpenrod doesn't make such a PR I will myself (I've done lots with LOTS in my own reseach so it'll be great to have it on the pi-Base).

StevenClontz commented:
Actually @keithpenrod may prefer to amend this PR to just have LOTS stuff. Either way let's keep this open until P000133 is added one way or another.

keithpenrod commented:
Thanks everyone for the feedback. I have removed P134 and all related theorems. I kept T273 and added a new theorem:
T274: discrete implies LOTS
I also removed all of the explicit declarations of P133 that I believe can be deduced.

I would like to do the other results as well, but as pointed out it might be best to preview what it looks like with this and see which results would be meaningful to add.

prabau commented on theorems/T000274.md:
```diff
@@ -0,0 +1,16 @@
+
+Since discrete spaces are homeomorphic precisely when they have the same cardinality, it suffices to show that for every cardinality $k$ there is a set of cardinality $k$ with a total order that induces the discrete topology.
+
+Let $k$ be a cardinal number. If $k$ is finite, then any total order on $k$ induces the discrete topology. If $k$ is infinite, let $X=\alpha \times \mathbb{Z}$ where $\alpha$ is an ordinal of cardinality $k$, together with the lexicographical ordering. Then we see that $X$ with the order topology is discrete. 
```
Nice proof.  Can we use $\kappa$ instead of $k$?

prabau commented on theorems/T000274.md:
```diff
@@ -0,0 +1,16 @@
+
+Let $k$ be a cardinal number. If $k$ is finite, then any total order on $k$ induces the discrete topology. If $k$ is infinite, let $X=\alpha \times \mathbb{Z}$ where $\alpha$ is an ordinal of cardinality $k$, together with the lexicographical ordering. Then we see that $X$ with the order topology is discrete. 
+
+See example 39 of {{doi:10.1007/978-1-4612-6290-9_6}}.
```
I don't see anything in S&S counterexample 39 that mentions the word discrete.  But this does for example: https://math.stackexchange.com/questions/3429508/verification-of-basic-properties-of-order-topology (see the answer from Mirko at the bottom). Need to fix the refs above also.

keithpenrod replied:
You're right. It's actually example 44.

prabau commented on spaces/S000038/properties/P000133.md:
```diff
@@ -0,0 +1,10 @@
+property: P000133
+value: true
+refs:
+- mr: '3728284'
+  name: Topology (Munkres)
```
It's not clear to use Munkres here.  Can you replace it with S&S, which is the place where we take the definition from?  And same for all the other files that use "By definition" if they come from S&S.

StevenClontz commented:
Thanks for your review @prabau. I implemented the changes you suggested - going to eyeball how things look in a dev version of the web interface then merge.

keithpenrod commented on theorems/T000274.md:
```diff
@@ -5,12 +5,12 @@ if:
 Let $\kappa$ be a cardinal number. If $\kappa$ is finite, then any total order on $k$ induces the discrete topology. If $\kappa$ is infinite, let $X=\alpha \times \mathbb{Z}$ where $\alpha$ is an ordinal of cardinality $\kappa$, together with the lexicographical ordering. Then we see that $X$ with the order topology is discrete. 
 
-See example 39 of {{doi:10.1007/978-1-4612-6290-9_6}}.
+See example 39 of {{mathse:3429751}}.
```
example 39 of that mathse reference doesn't make sense.

StevenClontz replied:
Whoops, thanks. #143

keithpenrod commented:
> Added later: Just realized what I have above is not right. Should be (LOTS + connected + separable ==> second countable), I guess.

Is connected needed, though? The sketch of the proof I have is if S is a countable dense set in the LOTS X then intervals of the form (a, b) where a, b in S\cup {-\infty, \infty} form a countable base for X. Does that not form a base if X is not connected?
